repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java

```diff
@@ -1351,23 +1347,66 @@ public void createAndQueueTask(String taskType, AtlasEdge edge) {
             return;
         }
 
+
+        AtlasVertex referenceVertex = GraphHelper.getPropagatingVertex(edge);
+        if(referenceVertex == null) {
+            return;
+        }
+
         List<AtlasVertex> currentClassificationVertices = GraphHelper.getPropagatableClassifications(edge);
         for (AtlasVertex currentClassificationVertex : currentClassificationVertices) {
+            String currentClassificationId = currentClassificationVertex.getIdForDisplay();
+            if(skipClassificationTaskCreation(currentClassificationId)) {
+                LOG.info("Task is already scheduled for classification id {}, no need to schedule task for edge {}", currentClassificationId, edge.getIdForDisplay());
+                continue;
+            }
 
-            AtlasVertex referenceVertex = GraphHelper.getPropagatingVertex(edge);
-            if(referenceVertex == null) {
-                return;
+            boolean removePropagationOnEntityDelete = GraphHelper.getRemovePropagations(currentClassificationVertex);
+
+            if (!(isTermEntityEdge || removePropagationOnEntityDelete)) {
+                LOG.debug("This edge is not term edge or remove propagation isn't enabled");
+                continue;
             }
```

Comment:
Please move this code block before the previous if statement
Why? 
imagine f (!(isTermEntityEdge || removePropagationOnEntityDelete))  is already resulting into true then we will save on making ES query

Reply:
Yes, you are right. Changed this